File: threads/src/main/java/im/threads/view/ChatActivity.java
package im.threads.view;

import android.os.Bundle;

import androidx.annotation.Nullable;

import im.threads.R;
import im.threads.internal.activities.BaseActivity;

/**
 * Вся логика находится во фрагменте. Смотрите {@link ChatFragment}
 */
public final class ChatActivity extends BaseActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_chat_activity);
    }

    @Override
    public void onBackPressed() {
        final ChatFragment chatFragment = (ChatFragment) getSupportFragmentManager().findFragmentByTag("frag_chat");
        if (chatFragment != null && chatFragment.onBackPressed()) {
            super.onBackPressed();
        }
    }
}

File: threads/src/main/java/im/threads/ui/activities/ChatActivity.kt
package im.threads.ui.activities

import android.content.pm.PackageManager
import android.os.Bundle
import android.widget.Toast
import im.threads.R
import im.threads.ui.fragments.ChatFragment

/**
 * Вся логика находится во фрагменте [ChatFragment]
 */
class ChatActivity : BaseActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.ecc_activity_chat_activity)
    }

    override fun onBackPressed() {
        (supportFragmentManager.findFragmentByTag("frag_chat") as? ChatFragment)?.let { chatFragment ->
            if (chatFragment.onBackPressed()) {
                super.onBackPressed()
            }
        }
    }

    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String?>, grantResults: IntArray) {
        if (requestCode == 123 && hasAllPermissionsGranted(grantResults)) {
            Toast.makeText(this, "All permissions are granted", Toast.LENGTH_LONG).show()
        } else {
            Toast.makeText(this, "Permissions not granted. Code: $requestCode", Toast.LENGTH_LONG).show()
        }
    }

    private fun hasAllPermissionsGranted(grantResults: IntArray): Boolean {
        for (grantResult in grantResults) {
            if (grantResult == PackageManager.PERMISSION_DENIED) {
                return false
            }
        }
        return true
    }
}
